crowdfunding_compassion/controllers/projects_controller.py

```diff
@@ -7,31 +7,77 @@
 #    The licence is in the file __manifest__.py
 #
 ##############################################################################
-from odoo import _
+
+import werkzeug
+from datetime import date, datetime
+
+from odoo import _, http, models
 from odoo.http import request, route, Controller, local_redirect
 from odoo.addons.cms_form.controllers.main import FormControllerMixin
 
 from ..forms.project_creation_form import NoGoalException,\
     NegativeGoalException, InvalidDateException, InvalidLinkException
+from odoo.addons.http_routing.models.ir_http import slug
+from odoo.addons.website.models.ir_http import sitemap_qs2dom
 
 
 class ProjectsController(Controller, FormControllerMixin):
-    @route("/projects", auth="public", website=True)
-    def get_projects_list(self, type=None, **kwargs):
-        project_obj = request.env["crowdfunding.project"]
+    _project_post_per_page = 12
 
-        # TODO connect pagination to backend -> CO-3213
-        return request.render(
-            "crowdfunding_compassion.project_list_page",
-            {"projects": project_obj.sudo().get_active_projects(type=type)},
-        )
+    def sitemap_projects(env, rule, qs):
+        projects = env['crowdfunding.project']
+        dom = sitemap_qs2dom(qs, '/projects', projects._rec_name)
+        dom += request.website.website_domain()
+        for f in projects.search(dom):
+            loc = '/project/%s' % slug(f)
+            if not qs or qs.lower() in loc:
+                yield {'loc': loc}
+
+    @route(["/projects", "/projects/page/<int:page>"], auth="public", website=True, sitemap=sitemap_projects)
+    def get_projects_list(self, type=None, page=1, year=None, **opt):
+        if request.website:
+            website_id = request.website.id
+            if website_id != 1:  # only for other site
```

Comment:
This way we match precisely the website
```suggestion
            if website_id == request.env.ref(
                    "crowdfunding_compassion.crowdfunding_website").sudo().id
```